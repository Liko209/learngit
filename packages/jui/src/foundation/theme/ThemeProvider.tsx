/*
 * @Author: Jeffrey Huang(jeffrey.huang@ringcentral.com)
 * @Date: 2018-08-30 08:39:11
 * Copyright Â© RingCentral. All rights reserved.
 */

// @ts-ignore
window.__MUI_USE_NEXT_TYPOGRAPHY_VARIANTS__ = true;
export { default as ThemeProvider } from 'rcui/foundation/styles/ThemeProvider';
