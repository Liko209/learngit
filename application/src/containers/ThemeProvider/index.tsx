import * as React from 'react';
import { ThemeProvider as UIThemeProvider } from 'ui-components/theme';
import { defaultTheme } from '@/theme';

interface IProps {
  children: React.ReactNode;
}
class ThemeProvider extends React.PureComponent<IProps> {
  render() {
    const { children } = this.props;
    return <UIThemeProvider themeName={defaultTheme}>{children}</UIThemeProvider>;
  }
}

export default ThemeProvider;
