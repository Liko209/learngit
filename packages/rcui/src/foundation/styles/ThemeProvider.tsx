import React, { Component } from 'react';
import { MuiThemeProvider } from '@material-ui/core/styles';
import { ThemeProvider as StyledThemeProvider } from '../styled-components';
import createTheme from './createTheme';
import themeHandler from './ThemeHandler';
import options from './options.json';
import { Theme } from './theme.d';

type ThemeProviderProps = {
  themeName?: string;
  children: React.ReactNode;
};

type ThemeProviderPropsState = {
  theme: Theme | null;
};

class ThemeProvider extends Component<
  ThemeProviderProps,
  ThemeProviderPropsState
> {
  private _mounted: boolean;
  constructor(props: ThemeProviderProps) {
    super(props);
    this.state = {
      theme: null,
    };
    this.onThemeChanged = this.onThemeChanged.bind(this);
  }

  async componentDidMount() {
    this._mounted = true;
    const { themeName } = this.props;
    let themeOptions = {};
    if (themeName) {
      themeOptions = await themeHandler.read(themeName);
    }
    this.setState({
      theme: createTheme(themeOptions),
    });

    themeHandler.on(options.bindEvent, this.onThemeChanged);
  }

  componentWillUnmount() {
    this._mounted = false;
    themeHandler.offAny(this.onThemeChanged);
  }

  onThemeChanged(themeOptions: {}) {
    if (!this._mounted) return;
    this.setState({
      theme: createTheme(themeOptions),
    });
  }

  render() {
    const { theme } = this.state;
    const { children } = this.props;
    if (!theme) {
      return null;
    }

    return (
      <MuiThemeProvider theme={theme}>
        <StyledThemeProvider theme={theme}>
          {React.Children.only(children)}
        </StyledThemeProvider>
      </MuiThemeProvider>
    );
  }
}

export default ThemeProvider;
