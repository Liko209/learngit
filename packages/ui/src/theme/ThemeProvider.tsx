import React from 'react';
import createMuiTheme, {
  ThemeOptions,
  Theme as MuiTheme,
} from '@material-ui/core/styles/createMuiTheme';
import { Palette as MuiPalette } from '@material-ui/core/styles/createPalette';
import MuiThemeProvider from '@material-ui/core/styles/MuiThemeProvider';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';

export interface ThemeProviderProps {
  themeName: string;
  children: React.ReactNode;
}

const createTheme = (options: ThemeOptions) =>
  createMuiTheme(options);

const createThemes = () => {
  const requireContext = require.context('./config', false, /.ts$/);
  const themePaths = requireContext.keys();

  return themePaths
    .reduce(
      (themes, themePath) => {
        const themeName = themePath.split('.')[1].split('/')[1];
        themes[themeName] = createTheme(requireContext(themePath)[themeName]);
        return themes;
      },
      {},
    );
};

const themes = createThemes();

const ThemeProvider: React.SFC<ThemeProviderProps> = ({ themeName, children }) => {
  const theme = themes[themeName];
  return (
    <StyledThemeProvider theme={theme}>
      <MuiThemeProvider theme={theme}>
        {React.Children.only(children)}
      </MuiThemeProvider>
    </StyledThemeProvider>
  );
};

type Theme = MuiTheme & {
  palette: MuiPalette & {
    primary: {
      900: string;
      800: string;
      700: string;
      600: string;
      500: string;
      400: string;
      300: string;
      200: string;
      100: string;
      50: string;
    },
    secondary: {
      900: string;
      800: string;
      700: string;
      600: string;
      500: string;
      400: string;
      300: string;
      200: string;
      100: string;
      50: string;
    };
    semantic: {
      negative: string;
      positive: string;
      critical: string;
      neutral: string;
    },
    accent: {
      ash: string;
      lake: string;
      tiffany: string;
      cateye: string;
      grass: string;
      gold: string;
      persimmon: string;
      tomato: string;
    };
  };
};

export {
  Theme,
  ThemeProvider,
};
