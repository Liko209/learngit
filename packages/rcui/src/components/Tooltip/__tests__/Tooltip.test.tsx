/*
 * @Author: Wayne Zhou (wayne.zhou@ringcentral.com)
 * @Date: 2019-03-21 22:55:09
 * Copyright Â© RingCentral. All rights reserved.
 */

import * as React from 'react';
import { RuiTooltip } from '../index';
import ThemeProvider from '../../../foundation/styles/ThemeProvider';
import * as renderer from 'react-test-renderer';

describe('Tooltip', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <RuiTooltip title="children">
            <div>children</div>
          </RuiTooltip>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
