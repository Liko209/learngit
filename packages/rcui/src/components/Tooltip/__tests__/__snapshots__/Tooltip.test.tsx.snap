// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders correctly 1`] = `
<div
  aria-describedby={null}
  className=""
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title={null}
>
  children
</div>
`;
